deployments/hadoop-yarn/ansible/38-install-user-db.yml

```diff
@@ -0,0 +1,150 @@
+#   <meta:licence>
+#     Copyright (c) 2022, ROE (http://www.roe.ac.uk/)
+#
+#     This information is free software: you can redistribute it and/or modify
+#     it under the terms of the GNU General Public License as published by
+#     the Free Software Foundation, either version 3 of the License, or
+#     (at your option) any later version.
+#
+#     This information is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU General Public License for more details.
+#
+#     You should have received a copy of the GNU General Public License
+#     along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#   </meta:licence>
+# </meta:header>
+#
+#
+
+---
+- name: "Install MySQL Database for Zeppelin/Shiro"
+  hosts: zeppelin
+  become: yes
+  become_user: root
+  gather_facts: false
+  vars_files:
+    - config/ansible.yml
+    - config/hadoop.yml
+    - /tmp/ansible-vars.yml
+    - config/zeppelin.yml
+  vars:
+    zeppelinshiro: |
+            [main]
+            ds = com.mysql.cj.jdbc.MysqlDataSource
+            ds.serverName = localhost
+            ds.databaseName = zeppelin
+            ds.user = zeppelin
+            ds.password = {{ mysql_zeppelin_password }}
+
+            jdbcRealm = org.apache.shiro.realm.jdbc.JdbcRealm
+            ps = org.apache.shiro.authc.credential.DefaultPasswordService
+            pm = org.apache.shiro.authc.credential.PasswordMatcher
+            pm.passwordService = $ps
+
+            jdbcRealm.dataSource = $ds
+            jdbcRealm.credentialsMatcher = $pm
+            shiro.loginUrl = /api/login
+
+            sessionManager = org.apache.shiro.web.session.mgt.DefaultWebSessionManager
+            cookie = org.apache.shiro.web.servlet.SimpleCookie
+            cookie.name = JSESSIONID
+            cookie.httpOnly = true
+            sessionManager.sessionIdCookie = $cookie
+            securityManager.sessionManager = $sessionManager
+            securityManager.sessionManager.globalSessionTimeout = 86400000
+
+
+            [urls]
+            /** = authc
+            /api/version = anon
+            /api/interpreter/setting/restart/** = authc
+            /api/interpreter/** = authc, roles[admin]
+            /api/configurations/** = authc, roles[admin]
+            /api/credential/** = authc, roles[admin]
+
+            [roles]
+            user = *
+            admin = *
+
+  tasks:
+
+   - set_fact:
+       mysql_root_password: "{{ lookup('password','/dev/null chars=ascii_letters,digits,hexdigits length=20') }}!"
+       mysql_zeppelin_password: "{{ lookup('password','/dev/null chars=ascii_letters,digits,hexdigits length=20') }}!"
+
+   - name: "Enable MySQL Community release repo"
+     yum:
+       disable_gpg_check: True
+       name: https://repo.mysql.com//mysql80-community-release-fc31-1.noarch.rpm
+       state: present
```

Comment:
If we install MariaDB rather than MySQL then it comes as part of the OS packages and we don't have to install the separate repository.